server.js

```diff
@@ -1,44 +1,55 @@
 const spawn = require('child_process').spawn;
 const express = require('express');
 const app = express();
+const server = require('http').Server(app);
+const io = require('socket.io')(server);
 const robot = require('robotjs');
 const request = require('request');
 const creds = require('../credentials.js');
-
+var bodyParser = require('body-parser')
 
 let currentBrowserProc = null;
 const YOUTUBE_KEY = creds.YOUTUBE_KEY;
 
 app.use(express.static('dist'));
 app.use(express.static('client'));
+app.use(bodyParser.json());
 
 app.get('/api/search/', (req, res) => {
-    let url = `https://www.googleapis.com/youtube/v3/search?key=${YOUTUBE_KEY}&part=snippet&q=${req.query.q}`;
-    request(url, (e, r, body) => res.send(body));
+    let url = 'https://www.googleapis.com/youtube/v3/search';
+    let query = `?key=${YOUTUBE_KEY}&part=snippet&q=${req.query.q}&type=video`;
+
+    request(url + query, (e, r, body) => {
+        let results = JSON.parse(body)
+                          .items
+                          .map(item => ({ytId: item.id.videoId, title: item.snippet.title}));
+        res.send(results);
+    });
 });
 
-
-app.get('/api/yt-stream/:url', (req, res) => {
+app.post('/api/play/', (req, res) => {
     if (currentBrowserProc) {
         currentBrowserProc.kill();
     }
 
-    currentBrowserProc = spawn('chromium-browser', [`https://www.youtube.com/watch?v=${req.params.url}`]);
+    currentBrowserProc = spawn('chromium-browser', [`https://www.youtube.com/watch?v=${req.body.ytId}`]);
     setTimeout(() => robot.keyTap('f'), 15000);
-});
 
+    io.emit('currently-playing', {title: req.body.title});
```

Comment:
I don't think that the server should depend on the client state. Lets the server holds a cache and use it to find the video from a ID. It will make lazy request for the videos outside of the cache